File: arrays/sliding-window/209. Minimum Size Subarray Sum/Solution.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        int minLength = int.MaxValue;
        for(int right = 0, left = 0, sum = 0; right < nums.Length; right++){
             sum += nums[right];
             while (sum >= target) {
                minLength = Math.Min(minLength, right-left+1);
                sum -= nums[left++];
             }
        }
        return minLength == int.MaxValue ? 0 : minLength;
    }
}

File: arrays/sliding-window/209. Minimum Size Subarray Sum/solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_length = float('inf')
        window_sum = 0
        left = 0
        for right, num in enumerate(nums):
             window_sum += num
             while window_sum >= target:
                min_length = min(min_length, right-left+1)
                window_sum -= nums[left]
                left += 1

        return min_length if min_length != float('inf') else 0
        
